Mage.Sets/src/mage/cards/e/EowynShieldmaiden.java

```diff
@@ -0,0 +1,141 @@
+package mage.cards.e;
+
+import java.util.*;
+
+import mage.MageInt;
+import mage.abilities.Ability;
+import mage.abilities.TriggeredAbility;
+import mage.abilities.common.BeginningOfCombatTriggeredAbility;
+import mage.abilities.condition.Condition;
+import mage.abilities.condition.common.PermanentsOnTheBattlefieldCondition;
+import mage.abilities.decorator.ConditionalInterveningIfTriggeredAbility;
+import mage.abilities.decorator.ConditionalOneShotEffect;
+import mage.abilities.effects.common.CreateTokenEffect;
+import mage.abilities.effects.common.DrawCardSourceControllerEffect;
+import mage.constants.*;
+import mage.abilities.keyword.FirstStrikeAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.filter.FilterPermanent;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.permanent.Permanent;
+import mage.game.permanent.token.HumanKnightToken;
+import mage.watchers.Watcher;
+
+/**
+ *
+ * @author Susucr
+ */
+public final class EowynShieldmaiden extends CardImpl {
+
+    private static final FilterPermanent filter = new FilterPermanent(SubType.HUMAN, "Humans");
+
+    public EowynShieldmaiden(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{2}{U}{R}{W}");
+        
+        this.supertype.add(SuperType.LEGENDARY);
+        this.subtype.add(SubType.HUMAN);
+        this.subtype.add(SubType.KNIGHT);
+        this.power = new MageInt(5);
+        this.toughness = new MageInt(4);
+
+        // First strike
+        this.addAbility(FirstStrikeAbility.getInstance());
+
+        // At the beginning of combat on your turn,
+        // if another Human entered the battlefield under your control this turn,
+        // create two 2/2 red Human Knight creature tokens with trample and haste.
+        // Then if you control six or more Humans, draw a card.
+
+        TriggeredAbility triggeredAbility = new BeginningOfCombatTriggeredAbility(
+            Zone.BATTLEFIELD,
+            new CreateTokenEffect(new HumanKnightToken(), 2),
+            TargetController.YOU, false, false
+        );
+        triggeredAbility.addEffect(new ConditionalOneShotEffect(
+            new DrawCardSourceControllerEffect(1),
+            new PermanentsOnTheBattlefieldCondition(filter, ComparisonType.MORE_THAN, 5)
```

Comment:
Need to make sure this checks controlled Humans, not any Humans

Reply:
I can add a true boolean as 4th argument, but the PermanentsOnTheBattlefieldCondition constructor with 3 arguments is already checking only controlled permanents.

https://github.com/magefree/mage/blob/7a33ca812ce37ecd2651b34aee95e3b95b4be05c/Mage/src/main/java/mage/abilities/condition/common/PermanentsOnTheBattlefieldCondition.java#L49C8-L59

Reply:
Ah never mind then, sorry